airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py

```diff
@@ -1163,3 +1166,163 @@ def stream_slices(
+            unique_records += records
+            reports = [{"report_id": report_id,
+                        "createdSince": start_date.strftime(DATE_TIME_FORMAT),
+                        "createdUntil": end_date.strftime(DATE_TIME_FORMAT)
+                        } for report_id in records]
+
+            yield from reports
+
+            next_value = response.get("nextToken", None)
+            params = {"nextToken": next_value}
+            if not next_value:
+                complete = True
+
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        """
+        Create and retrieve the report.
+        Decrypt and parse the report is its fully proceed, then yield the report document records.
+        """
+        report_payload = {}
+        is_processed = False
+        is_done = False
+        start_time = pendulum.now("utc")
+        seconds_waited = 0
+        report_id = self._create_report(sync_mode, cursor_field, stream_slice, stream_state)["reportId"]
+
+        # create and retrieve the report
+        while not is_processed and seconds_waited < self.max_wait_seconds:
+            report_payload = self._retrieve_report(report_id=report_id)
+            seconds_waited = (pendulum.now("utc") - start_time).seconds
+            processing_status = report_payload.get("processingStatus")
+
+            is_processed = ProcessingStatus.is_processed(processing_status)
+            is_done = processing_status == ProcessingStatus.DONE
+            is_cancelled = processing_status == ProcessingStatus.CANCELLED
+            is_fatal = processing_status == ProcessingStatus.FATAL
+            
+            time.sleep(self.sleep_seconds)
+        
+        if is_done:
+            # retrieve and decrypt the report document
+            document_id = report_payload["reportDocumentId"]
+            request_headers = self.request_headers()
+            request = self._create_prepared_request(
+                path=self.path(document_id=document_id),
+                headers=dict(request_headers, **self.authenticator.get_auth_header()),
+                params=self.request_params(),
+            )
+            response = self._send_request(request)
+            
+            for r in self.parse_response(response, stream_state, stream_slice):
+                yield r
+
+            new_cursor_value = stream_slice.get("createdUntil")
+            if (self._cursor_value is None) or (pendulum.parse(new_cursor_value) > pendulum.parse(self._cursor_value)):
+                self._cursor_value =  new_cursor_value
+                logger.info(f"Updating cursor value for {self.name} stream to {self._cursor_value}")
+            
+
+        elif is_fatal:
+            raise Exception(f"The report for stream '{self.name}' was aborted due to a fatal error")
```

Comment:
What I said here is, if there is a fatal error from the API please log it to the user understand what caused it